### PR TITLE
AVX-23422: Not allow enable custom algorithm with default value for all fields

### PR DESCRIPTION
**Pull request by zhiqiangjin**
Same checks as site2cloud.

### CODE REVIEW

**Review by shuoliugit — DISMISSED** [review]

File: aviatrix/resource_aviatrix_spoke_external_device_conn.go [hunk 1]
```diff
@@ -396,10 +396,29 @@ func resourceAviatrixSpokeExternalDeviceConnCreate(d *schema.ResourceData, meta
...
+			externalDeviceConn.Phase2Encryption == "" {
+			return fmt.Errorf("custom_algorithms is enabled, please set all of the algorithm parameters")
+		} else if externalDeviceConn.Phase1Auth == goaviatrix.Phase1AuthDefault &&
+			externalDeviceConn.Phase2Auth == goaviatrix.Phase2AuthDefault &&
+			externalDeviceConn.Phase1DhGroups == goaviatrix.Phase1DhGroupDefault &&
+			externalDeviceConn.Phase2DhGroups == goaviatrix.Phase2DhGroupDefault &&
+			externalDeviceConn.Phase1Encryption == goaviatrix.Phase1EncryptionDefault &&
+			externalDeviceConn.Phase2Encryption == goaviatrix.Phase2EncryptionDefault {
+			return fmt.Errorf("custom_algorithms is enabled, cannot use default values for " +
+				"all six algorithm parameters. Please change value of one or multiple of the six algorithm parameters")
```
Comment (mlin-aviatrix) [inline aviatrix/resource_aviatrix_spoke_external_device_conn.go]: Maybe the last part can be reworded to `Please change the value of at least one of the six algorithm parameters`

File: aviatrix/resource_aviatrix_spoke_external_device_conn.go [hunk 2]
```diff
@@ -396,10 +396,29 @@ func resourceAviatrixSpokeExternalDeviceConnCreate(d *schema.ResourceData, meta
...
+		}
+	} else {
+		if externalDeviceConn.Phase1Auth != "" || externalDeviceConn.Phase1DhGroups != "" ||
+			externalDeviceConn.Phase1Encryption != "" || externalDeviceConn.Phase2Auth != "" ||
+			externalDeviceConn.Phase2DhGroups != "" || externalDeviceConn.Phase2Encryption != "" {
+			return fmt.Errorf("custom_algorithms is not enabled, all algorithms fields should be left empty")
```
Comment (mlin-aviatrix) [inline aviatrix/resource_aviatrix_spoke_external_device_conn.go]: "algorithms" should be singular. Should be `custom_algorithms is not enabled, all algorithm fields should be left empty`

File: aviatrix/resource_aviatrix_transit_external_device_conn.go [hunk 3]
```diff
@@ -480,10 +480,29 @@ func resourceAviatrixTransitExternalDeviceConnCreate(d *schema.ResourceData, met
...
+			externalDeviceConn.Phase2Encryption == "" {
+			return fmt.Errorf("custom_algorithms is enabled, please set all of the algorithm parameters")
+		} else if externalDeviceConn.Phase1Auth == goaviatrix.Phase1AuthDefault &&
+			externalDeviceConn.Phase2Auth == goaviatrix.Phase2AuthDefault &&
+			externalDeviceConn.Phase1DhGroups == goaviatrix.Phase1DhGroupDefault &&
+			externalDeviceConn.Phase2DhGroups == goaviatrix.Phase2DhGroupDefault &&
+			externalDeviceConn.Phase1Encryption == goaviatrix.Phase1EncryptionDefault &&
+			externalDeviceConn.Phase2Encryption == goaviatrix.Phase2EncryptionDefault {
+			return fmt.Errorf("custom_algorithms is enabled, cannot use default values for " +
+				"all six algorithm parameters. Please change value of one or multiple of the six algorithm parameters")
```
Comment (mlin-aviatrix) [inline aviatrix/resource_aviatrix_transit_external_device_conn.go]: Same.

File: aviatrix/resource_aviatrix_transit_external_device_conn.go [hunk 4]
```diff
@@ -480,10 +480,29 @@ func resourceAviatrixTransitExternalDeviceConnCreate(d *schema.ResourceData, met
...
+		}
+	} else {
+		if externalDeviceConn.Phase1Auth != "" || externalDeviceConn.Phase1DhGroups != "" ||
+			externalDeviceConn.Phase1Encryption != "" || externalDeviceConn.Phase2Auth != "" ||
+			externalDeviceConn.Phase2DhGroups != "" || externalDeviceConn.Phase2Encryption != "" {
+			return fmt.Errorf("custom_algorithms is not enabled, all algorithms fields should be left empty")
```
Comment (mlin-aviatrix) [inline aviatrix/resource_aviatrix_transit_external_device_conn.go]: Same.

**Review by mlin-aviatrix — CHANGES_REQUESTED** [review]

**Review by mlin-aviatrix — APPROVED** [review]

**Pull request merged**